src/spherical_mesh.cc

```diff
@@ -50,14 +41,18 @@ double SphericalMesh::volume(int i) const
     return (4.*M_PI/3.)*(r_out*r_out*r_out - r_in*r_in*r_in);
     }
 
-double SphericalMesh::gradient(int idx, double f_lo, double f_hi) const
+double SphericalMesh::gradient(int /*i*/, double f_lo, double f_hi) const
     {
-    if (idx == 0 && origin_ == 0.){
-        return 0;
-        }
-    else{
-        return (f_hi-f_lo)/(step_);
-        }
+    return (f_hi-f_lo)/(step_);
     }   
-
+    
+void SphericalMesh::validateBoundaryCondition() const
+    {
+    Mesh::validateBoundaryCondition();
+    
+    if(lower_bc_ == BoundaryType::periodic || upper_bc_ == BoundaryType::periodic || upper_bc_ == BoundaryType::reflect)
+        {
+         throw std::invalid_argument("Periodic boundary conditions or reflect upper boundary condition is  not valid in spherical geometry");
+        }
```

Comment:
Make these separate so the message can be more specific
```suggestion
    if(lower_bc_ == BoundaryType::periodic || upper_bc_ == BoundaryType::periodic)
        {
         throw std::invalid_argument("Periodic boundary conditions invalid in spherical geometry");
        }
    else if  (upper_bc_ == BoundaryType::reflect)
        {
        throw std::invalid_argument("Reflect boundary condition invalid for upper boundary");
        }
```